homeworkLecturesW10D03/src/ba/bitcamp/Files/MyPanel.java

```diff
@@ -0,0 +1,181 @@
+
+	// Creating scroll pane
+	private JScrollPane scrollPane = new JScrollPane(textArea,
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+
+	/**
+	 * Constructor for creating MyPanel object with text area, menu bar and
+	 * scroll pane
+	 */
+	public MyPanel() {
+
+		// Seting panel's layout
+		setLayout(new BorderLayout());
+
+		// Adding menu bar to the north of panel
+		add(menuBar, BorderLayout.NORTH);
+
+		// Adding menu to the menu bar
+		menuBar.add(menu);
+
+		// Adding menu items to the menu
+		menu.add(newFile);
+		menu.add(open);
+		menu.add(save);
+
+		// Making text area
+		textArea.setLineWrap(true);
+
+		// Adding scroll pane to the center of the panel
+		add(scrollPane, BorderLayout.CENTER);
+
+		/*
+		 * ActionListener for creating empty text file
+		 */
+		newFile.addActionListener(new ActionListener() {
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				textArea.setText("");
+			}
+		});
+
+		/*
+		 * ActionListener for opening existing text file
+		 */
+		open.addActionListener(new ActionListener() {
+
+			private BufferedReader br;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				try {
+					UIManager.setLookAndFeel(UIManager
+							.getSystemLookAndFeelClassName());
+				} catch (ClassNotFoundException | InstantiationException
+						| IllegalAccessException
+						| UnsupportedLookAndFeelException e1) {
+					e1.printStackTrace();
+				}
+				JFileChooser openFile = new JFileChooser();
+				int action = openFile.showOpenDialog(open);
+				if (action == JFileChooser.OPEN_DIALOG) {
+					File file = openFile.getSelectedFile();
+					try {
+						FileReader fr = new FileReader(file);
+						br = new BufferedReader(fr);
+						String line = br.readLine();
+						String text = line + "\n";
+						while (line != null) {
+							line = br.readLine();
+							text += line + "\n";
+						}
+						textArea.setText(text);
+					} catch (FileNotFoundException e1) {
+						e1.printStackTrace();
+					} catch (IOException e1) {
+						e1.printStackTrace();
+					}
+
+				}
+			}
+		});
+
+		/*
+		 * ActionListener for saving text file
+		 */
+		save.addActionListener(new ActionListener() {
+
+			private BufferedWriter bwr;
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				try {
+					UIManager.setLookAndFeel(UIManager
+							.getSystemLookAndFeelClassName());
+				} catch (ClassNotFoundException | InstantiationException
+						| IllegalAccessException
+						| UnsupportedLookAndFeelException e1) {
+					e1.printStackTrace();
+				}
+				JFileChooser saveFile = new JFileChooser();
+				FileNameExtensionFilter ext = new FileNameExtensionFilter(
+						".txt File", "txt");
+				saveFile.setFileFilter(ext);
+				int action = saveFile.showSaveDialog(save);
+
+				if (action == JFileChooser.APPROVE_OPTION) {
+					String text = textArea.getText();
+					text = text.replaceAll("(?!\\r)\\n", "\r\n");
```

Comment:
This could be inefficient in case large file is opened. Also, java is write once, use everywhere so new lines should be platform specific: `System.lineSeparator()`